tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1382,7 +1382,7 @@ bool DuckDBPyConnection::IsAcceptedArrowObject(const py::object &object) {
 	    py_import_cache.arrow_lib().RecordBatchReader.IsInstance(object)) {
 		return true;
 	}
-	if (!py_import_cache.arrow_dataset().IsLoaded()) {
+	if (!ModuleIsLoaded<ArrowDatasetCacheItem>()) {
```

Comment:
While this is a good change, as we don't unnecessarily load a module we don't need this way, it shouldn't be required to fix the issue.

Recently I pushed a fix for missing attributes in a module, by checking if the module has the attribute in question (hasattr).
IsInstance will still work if the object isn't loaded and just return false